velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -0,0 +1,306 @@
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/type/tests/FilterBuilder.h"
+#include "velox/type/tests/SubfieldFiltersBuilder.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+
+class ParquetTpchTest : public HiveConnectorTestBase {
+ protected:
+  // Setup a duckDB instance for the entire suite and load TPCH data with scale
+  // factor 0.01.
+  static void SetUpTestCase() {
+    if (duckDb_ == nullptr) {
+      duckDb_ = std::make_shared<DuckDbQueryRunner>();
+      constexpr double tpchsf = 0.01;
+      duckDb_->initTPCH(tpchsf);
+    }
+    functions::prestosql::registerAllFunctions();
+  }
+
+  void SetUp() override {
+    HiveConnectorTestBase::SetUp();
+    parquet::registerParquetReaderFactory();
+    tempDirectory_ = tempDirectory_ = exec::test::TempDirectoryPath::create();
+  }
+
+  void TearDown() override {
+    parquet::unregisterParquetReaderFactory();
+    HiveConnectorTestBase::TearDown();
+  }
+
+  int64_t date(const std::string_view string_date) const {
+    Date date;
+    parseTo(string_date, date);
+    return date.days();
+  }
+
+  // For a given file at filePath, add splits equal to numSplits.
+  std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> makeSplits(
+      const std::string& filePath,
+      int64_t numSplits) const {
+    LocalReadFile lfs(filePath);
+    const int fileSize = lfs.size();
+    const int splitSize = fileSize / numSplits;
+    std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> splits;
+
+    // Add all the splits except the last one.
+    int i = 0;
+    for (; i < numSplits - 1; i++) {
+      auto split = makeHiveConnectorSplit(filePath, i * splitSize, splitSize);
+      split->fileFormat = dwio::common::FileFormat::PARQUET;
+      splits.push_back(std::move(split));
+    }
+    // Add the last split with the remaining as start.
+    auto split = makeHiveConnectorSplit(filePath, i * splitSize);
+    split->fileFormat = dwio::common::FileFormat::PARQUET;
+    splits.push_back(std::move(split));
+    return splits;
+  }
+
+  void addSplitsToTask(
+      exec::Task* task,
+      bool& noMoreSplits,
+      const std::vector<std::string>& filePaths,
+      int sourcePlanNodeId = 0,
+      int64_t numSplits = 10) const {
+    if (!noMoreSplits) {
+      for (const auto& filePath : filePaths) {
+        auto const& splits = makeSplits(filePath, numSplits);
+        for (const auto& split : splits) {
+          task->addSplit(std::to_string(sourcePlanNodeId), exec::Split(split));
+        }
+      }
+      task->noMoreSplits(std::to_string(sourcePlanNodeId));
+      noMoreSplits = true;
+    }
+  }
+
+  // Write the Lineitem TPCH table to a Parquet file and return the file
+  // location.
+  std::string writeDuckDBTPCHLineitemTableToParquet() const {
```

Comment:
This method name is very long and hard to read. Perhaps, add tableName argument to this method to remove "Lineitem" from the method name.

```
std::string writeTpchTableAsParquet(const std::string& tableName)
```